src/Aspire.Hosting/ContainerResourceBuilderExtensions.cs

```diff
@@ -138,18 +139,57 @@ public static IResourceBuilder<T> WithImageRegistry<T>(this IResourceBuilder<T>
     /// <param name="image">Image value.</param>
     /// <param name="tag">Tag value.</param>
     /// <returns></returns>
-    public static IResourceBuilder<T> WithImage<T>(this IResourceBuilder<T> builder, string image, string tag = "latest") where T : ContainerResource
+    public static IResourceBuilder<T> WithImage<T>(this IResourceBuilder<T> builder, string image, string? tag = null) where T : ContainerResource
```

Comment:
I've had to make an API breaking change here - specifically changing `string tag = "latest"` to `string? tag = null`.

Without this, `WithImage("foo:tag")` would become `WithImage("foo:tag", "latest")` which would be ambiguous.

Open to other suggestions

Reply:
If we are going to break it might be better to break it by removing the tag argument altogether.

Thoughts @davidfowl / @danegsta ?

Reply:
Would splitting the method up into two be a better option than changing the default value?
```cs
public static IResourceBuilder<T> WithImage<T>(this IResourceBuilder<T> builder, string image) where T : ContainerResource

// Possibly with addition of [Obsolete("Use 'WithImage(image).WithTag(tag) instead.")]
public static IResourceBuilder<T> WithImage<T>(this IResourceBuilder<T> builder, string image, string tag) where T : ContainerResource
    => builder.WithImage(image).WithImageTag(tag);
```

For my understanding, what exactly is the breaking change the public api analyser is complaining about here?  Is it the `string!` to `string?` - I'd have thought that would be OK as we're widening allowed inputs, not narrowing or is it the change of the default value itself.  Or is it the change of the default value from `"latest"` to `null`?  I know changing the default value is problematic for binary compatibility as anyone who previously called the method will still have the old `"latest"` default baked in, however in this particular case we ultimately set the tag to `"latest"` if the tag is `null` so the impact of changing the default should be mostly non observable.

Reply:
Any thoughts @davidfowl / @danegsta ?

Reply:
I like the idea of making the default tag null and we’d internally still use latest only if the tag wasn’t set. Feels like we should make this change now if we’re going to do it

Reply:
@davidfowl  Just to check, do you mean handle the default value within the `WithImage` method as done in the PR (i.e. setting ContainerImageAnnotation.Tag to `latest` method if the tag argument is null / missing), or do you mean handle the default elsewhere (i.e. leave `ContainerImageAnnotation.Tag` as null and then handling that later on.)

Reply:
probably as delayed as possible. Though I am concered at the number of places that might need to update the handling of a null tag.

Reply:
3 types of tests broke

1. Tests asserting that `ContainerImageAnnotation.Tag` was `latest` when they didn't pass a tag. (expected to break as that's what we changed)
2. Mainfests for images with tags no longer have `:latest` in theim  (i.e. `nginx:latest` has become just `nginx`)
3. Image in bicep files without a tag no longer have `:latest` on them.

Both 2 & 3 share the same root cause with `TryGetContainerImageName` which currently has explicitly logic to not include the tag if one isn't set  https://github.com/dotnet/aspire/blob/0b562b8eb039343df763d6e9083da8fbb938604e/src/Aspire.Hosting/ApplicationModel/ResourceExtensions.cs#L242-L246

This change also removes the `:latest` from the dashboard UI if no tag is specified:

![image](https://github.com/user-attachments/assets/6ac79ce0-6fff-4bd7-bc83-0be96f0754be)

```cs
builder.AddContainer("no-tag", "nginx");
builder.AddContainer("explicit-latest", "nginx:latest");
```

So the question is should aspire explicitly set the tag to latest before passing to the manifest / container runtime, or should it just leave the tag blank and let that system deal with it?  The PR currently has the latter, but changing the former should be a matter of changing `TryGetContainerImageName`, backing out the test changes from b93a3f2f59c8e0804e887222576e9ab32f3a6f26 and 81a77b3b245a6cffe16d778af6608be6a0ce0f95 and adding some additional test coverage.